spec/controllers/coronavirus_pages_controller_spec.rb

```diff
@@ -82,4 +83,48 @@
       expect(response).to redirect_to(coronavirus_pages_path)
     end
   end
+
+  describe "GET /coronavirus/:coronavirus_page_slug/reorder" do
+    before do
+      stub_user.permissions << "Unreleased feature"
+    end
+
+    it "renders page successfuly" do
+      get :reorder, params: { slug: coronavirus_page.slug }
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "POST /coronavirus/:coronavirus_page_slug/reorder" do
+    before do
+      stub_user.permissions << "Unreleased feature"
+      stub_request(:get, coronavirus_page.raw_content_url)
+        .to_return(status: 200, body: raw_content)
+      stub_coronavirus_publishing_api
+      live_stream
+    end
+    let!(:sub_section_0) { create :sub_section, position: 0, coronavirus_page: coronavirus_page }
+    let!(:sub_section_1) { create :sub_section, position: 1, coronavirus_page: coronavirus_page }
+
+    let(:section_params) { "[{\"id\":#{sub_section_0.id}, \"position\":1},{\"id\":#{sub_section_1.id},\"position\":0}]" }
```

Comment:
I think if `to_json` is used on an array, the construction of this data becomes a little simpler and easier to manipulate. For example, if `section_params` was built like this:
```ruby
let(:sub_section_0_params) { { id:  sub_section_0.id, position: 1 }  }
let(:sub_section_1_params) { { id:  sub_section_0.id, position: 0 }   }
let(:section_params) { [ sub_section_0_params, sub_section_1_params ].to_json }
```
an additional context could be added that clarified that it was the `position` data that was setting the position and that reorder wasn't simply doing a reverse sort.
``` ruby
context "with positions reversed" do
  let(:sub_section_0_params) { { id:  sub_section_0.id, position: 0 }  }
  let(:sub_section_1_params) { { id:  sub_section_0.id, position: 1 }   }

  it "reverses the positions" do
    subject
    expect(sub_section_0.reload.position).to eq 0
    expect(sub_section_1.reload.position).to eq 1
  end
end
```